src/Controller/vCardController.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Oveleon\ContaoCompanyBundle\Controller;
+
+use Contao\PageModel;
+use Oveleon\ContaoCompanyBundle\Generator\vCardGenerator;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @Route(defaults={"_scope" = "frontend"})
+ */
+class vCardController extends AbstractController
+{
+    private vCardGenerator $generator;
+
+    /**
+     * @param vCardGenerator $generator
+     */
+    public function __construct(vCardGenerator $generator)
+    {
+        $this->generator = $generator;
+    }
+
+    /**
+     * @Route("/company/vcard/download", name="contao_company_vcard_download")
+     *
+     * @param Request $request
+     *
+     * @return Response
+     *
+     * @throws \Exception
+     */
+    public function download(Request $request): Response
+    {
+        // Try to load the current contao page
+        $pageId = (int) $request->query->get('contaoPage');
```

Comment:
The parameter in this route doesn't match the one within the inserttag ('redirect').
I'd suggest changing it to something like 'page' in both files.

Reply:
Oh, I forgot to update the parameter after I changed it.